### PR TITLE
Add decoding for the full on chain data

### PR DESCRIPTION
**Pull request by sz-piotr**
The decoding for forced operations is still experimental and I expect it to break. Further analysis has to be done, to find a real world example and test on it, as the one we're using has 0 forced operations.

### CODE REVIEW

**Comment by sz-piotr** [comment]
@sin6pi7 The code now works with real data, care to review :) ?

File: packages/encoding/src/data/index.ts [hunk 1]
```diff
@@ -6,9 +6,10 @@ run()
 async function run() {
   const data = await getOnChainData(
     '0x46c212912be05a090a9300cf87fd9434b8e8bbca15878d070ba83375a5dbaebd'
+    // '0x5a81a54eb926a240264f9a481810242987ae566e82f29bcdd8cd32063e1976c5'
```
Comment (sin6pi7) [inline packages/encoding/src/data/index.ts:9]: Why is it commented out?

File: packages/encoding/test/readConditions.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { readToDecode } from './readToDecode'
+
+describe(readConditions.name, () => {
+  const decode = readToDecode(readConditions)
+
+  it('fails for empty data', () => {
+    expect(() => decode('')).toThrow(DecodingError, 'Went out of bounds')
+  })
+
+  it('can read zero conditions', () => {
```
Comment (sin6pi7) [inline packages/encoding/test/readConditions.test.ts:15]: IMO `can` word does not bring any value in test case title - if we made it `reads zero conditions` it says the same thing, but is shorter at the same time. Wdyt?

**Review by sin6pi7 — APPROVED** [review]
Only cosmetic changes requested - overall LGTM 👍

File: packages/encoding/src/data/index.ts (hunk 1 above)
Comment (sz-piotr) [reply packages/encoding/src/data/index.ts:9]: The first hash allows you to get the original data we had, the second is for getting data with forced actions

File: packages/encoding/test/readConditions.test.ts (hunk 2 above)
Comment (sz-piotr) [reply packages/encoding/test/readConditions.test.ts:15]: I am happy to change it but it is already like this everywhere. If you insist we can make a big update in another PR
image

Comment (sin6pi7) [reply packages/encoding/test/readConditions.test.ts:15]: Ok, makes sense - let's do it in another PR at some point.

**Pull request merged**